Bug: code generation breaks with relation

```prisma
datasource db {
	provider = "sqlite"
	url      = "file:dev.sqlite"
}

generator db {
	provider = "go run ../../.."
	output = "./db_gen.go"
	package = "relations"
}

model User {
	id       String  @id @default(cuid())
	email    String  @unique
	username String
	name     String?

	role     User?   @relation(fields: [roleID], references: [id])
	roleID   String?

	posts    Post[]
	comments Comment[]
}

model Post {
	id       String  @id @default(cuid())
	title    String
	content  String?

	author   User @relation(fields: [authorID], references: [id])
	authorID String

	comments Comment[]
}

model Comment {
	id        String  @id @default(cuid())
	content   String

	by        User @relation(fields: [userID], references: [id])
	userID    String

	post      Post @relation(fields: [postID], references: [id])
	postID    String
}

model Role {
	id   String @id @default(cuid())
	name String
}
```

results in error:

```
test/projects/relations/db_gen.go:1689:6: userToUserFindOne redeclared in this block
```

--- a/generator/templates/actions/find.gotpl
+++ b/generator/templates/actions/find.gotpl
@@ -13,9 +13,9 @@
 			{{ $relationName := $model.Name.GoCase }}
 
 			{{ if ne $field.Name "" }}
-				{{ $result = (print $name "To" $field.Type.GoCase "Find" $v.Name) }}
-				{{ $updateResult = (print $name "To" $field.Type.GoCase "Update" $v.Name) }}
-				{{ $deleteResult = (print $name "To" $field.Type.GoCase "Delete" $v.Name) }}
+				{{ $result = (print $name "To" $field.Name.GoCase "Find" $v.Name) }}
+				{{ $updateResult = (print $name "To" $field.Name.GoCase "Update" $v.Name) }}
+				{{ $deleteResult = (print $name "To" $field.Name.GoCase "Delete" $v.Name) }}
 				{{ $relationName = $field.Type.GoCase }}
 			{{ end }}
 

--- a/generator/templates/query.gotpl
+++ b/generator/templates/query.gotpl
@@ -95,8 +95,8 @@
 				{{ if $field.IsList }}
 					params ...i{{ $field.Type.GoCase }}Params,
 				{{ end }}
-			) {{ $name }}To{{ $field.Type.GoCase }}Find{{ if $field.IsList }}Many{{ else }}One{{ end }} {
-				var v {{ $name }}To{{ $field.Type.GoCase }}Find{{ if $field.IsList }}Many{{ else }}One{{ end }}
+			) {{ $name }}To{{ $field.Name.GoCase }}Find{{ if $field.IsList }}Many{{ else }}One{{ end }} {
+				var v {{ $name }}To{{ $field.Name.GoCase }}Find{{ if $field.IsList }}Many{{ else }}One{{ end }}
 
 				v.query.Operation = "query"
 				v.query.Method = "{{ $field.Name }}"

--- a/test/projects/relations/relations_test.go
+++ b/test/projects/relations/relations_test.go
@@ -67,7 +67,7 @@ func TestRelations(t *testing.T) {
 				t.Fatalf("fail %s", err)
 			}
 
-			expected := `{"id":"relations","email":"john@example.com","username":"johndoe","name":"John","posts":[{"id":"a","title":"common","content":"a","authorID":"relations","categoryID":null,"author":null,"Category":null,"comments":null},{"id":"b","title":"common","content":"b","authorID":"relations","categoryID":null,"author":null,"Category":null,"comments":null}],"comments":null}`
+			expected := `{"id":"relations","email":"john@example.com","username":"johndoe","name":"John","roleID":null,"role":null,"posts":[{"id":"a","title":"common","content":"a","authorID":"relations","categoryID":null,"author":null,"Category":null,"comments":null},{"id":"b","title":"common","content":"b","authorID":"relations","categoryID":null,"author":null,"Category":null,"comments":null}],"comments":null,"user":null}`
 			assert.Equal(t, expected, string(actual))
 		},
 	}, {

--- a/test/projects/relations/schema.prisma
+++ b/test/projects/relations/schema.prisma
@@ -15,6 +15,11 @@ model User {
 	username String
 	name     String?
 
+	// TODO separate this into an extra test
+	// related issue: https://github.com/prisma/prisma-client-go/issues/126
+	role     User?   @relation(fields: [roleID], references: [id])
+	roleID   String?
+
 	posts    Post[]
 	comments Comment[]
 }
@@ -30,6 +35,7 @@ model Post {
 	// 1) optional relation
 	// 2) uppercase model
 	// TODO create separate tests for these two cases
+	// related issue: https://github.com/prisma/prisma-client-go/issues/127
 	Category   Category?   @relation(fields: [categoryID], references: [id])
 	categoryID String?
 
@@ -48,6 +54,11 @@ model Comment {
 }
 
 model Category {
+	id   String @id @default(cuid())
+	name String
+}
+
+model Role {
 	id   String @id @default(cuid())
 	name String
 }

